tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_caches.py

```diff
@@ -382,13 +383,18 @@ def test_fill_to_local_k_caches_section_splitting_forward():
                 sections=[
                     VerticalLoopSectionFactory(
                         interval=IntervalFactory(end=AxisBound(level=LevelMarker.END, offset=-1)),
-                        horizontal_executions__0__body=[
-                            AssignStmtFactory(
-                                left__name="foo", right__name="foo", right__offset__k=0
-                            ),
-                            AssignStmtFactory(
-                                left__name="foo", right__name="foo", right__offset__k=1
-                            ),
+                        horizontal_executions=[
+                            HorizontalExecutionFactory(
+                                body=[
+                                    AssignStmtFactory(
+                                        left__name="foo", right__name="foo", right__offset__k=0
+                                    ),
+                                    AssignStmtFactory(
+                                        left__name="foo", right__name="foo", right__offset__k=1
+                                    ),
+                                ],
+                                declarations=[LocalScalarFactory()],
```

Comment:
Is this change related to the name clash? I cannot see why...

Reply:
Yes it is. The local scalar gets duplicated when the section is split. So we have a name clash if we do not rename one of the local scalars.

Reply:
makes sense.